lib/spack/spack/database.py

```diff
@@ -372,9 +372,27 @@ def _read_suppress_error():
             old_data = self._data
             try:
                 self._data = {}
+                missing_specs = set(directory_layout.all_specs())
```

Comment:
@tgamblin I am not sure I understood what you meant here. Can you give a look and see if I implemented the thing the way you were asking?